tests/Functional/StatementTest.php

```diff
@@ -236,6 +241,28 @@ public function testBindParamWithNullLength(): void
         self::assertEquals(1, $stmt->executeQuery()->fetchOne());
     }
 
+    public function testBindInvalidNamedParameter(): void
+    {
+        if (TestUtil::isDriverOneOf('ibm_db2', 'mysqli', 'sqlsrv')) {
+            self::markTestSkipped('The driver does not support named statement parameters');
+        }
+
+        if (PHP_VERSION_ID < 80000 && TestUtil::isDriverOneOf('pdo_oci')) {
+            self::markTestSkipped('pdo_oci on PHP 7 does not report this error');
+        }
+
+        $platform  = $this->connection->getDatabasePlatform();
+        $statement = $this->connection->prepare($platform->getDummySelectSQL(':foo'));
+        $this->expectException(DriverException::class);
+
+        // prevent the PHPUnit error handler from handling the warning that oci_bind_by_name() may trigger
+        if (TestUtil::isDriverOneOf('oci8')) {
+            $this->iniSet('error_reporting', (string) (E_ALL & ~E_WARNING));
```

Comment:
This will have to be replaced with a `@` before `oci_bind_by_name()` in `4.0.x` since the error there is reported as a driver exception.